script/tool/lib/src/common/package_command.dart

```diff
@@ -595,7 +595,15 @@ abstract class PackageCommand extends Command<void> {
     // ... and then check whether it has an enclosing package.
     final RepositoryPackage package = RepositoryPackage(currentDir);
     final RepositoryPackage? enclosingPackage = package.getEnclosingPackage();
-    return (enclosingPackage ?? package).directory.basename;
+    final RepositoryPackage rootPackage = enclosingPackage ?? package;
+    final String name = rootPackage.directory.basename;
+    // For an app-facing package in a federated plugin, return the fully
+    // qualified name, since returning just the name will cause the entire
+    // group to run.
+    if (rootPackage.directory.parent.basename == name) {
```

Comment:
Couldn't you just always return `rootPackage.directory.parent.basename/name` instead of having this extra logic?

Reply:
No, because for, e.g., `pigeon` that would be `packages/pigeon` which doesn't work.

(I could always do it for federated plugins, but that ends up being about the same amount of logic.)